packages/svelte/types/index.d.ts

```diff
@@ -621,6 +621,19 @@ declare module 'svelte/animate' {
 	export {};
 }
 
+declare module 'svelte/attachments' {
+	/**
+	 * Creates a `Symbol` that Svelte recognises as an attachment key
+	 */
+	export function createAttachmentKey(): symbol;
+	/**
+	 * Returns `true` if the symbol was created with `createAttachmentKey`
+	 * */
+	export function isAttachmentKey(key: string | symbol): boolean;
```

Comment:
```suggestion
	export function isAttachmentKey(key: string | symbol): key is <Attachment Type>;
```

Would this be possible?  This way the function can immediately be used for type narrowing.  Also:  Is there such a thing as an attachment type? Would be ideal so we can type properties.

Reply:
You'd have to narrow the property value (which will be the actual attachment function), not the property key.
Besides, you can't be sure that the property value was actually an attachment function, someone could've done:
```js
  const stuff = {
    [createAttachmentKey()]: 'not actually an attachment, lol'
  };
```
for all you know

Reply:
Yes, it was very late last night.  I was thinking about a `isAttachment` function.  Brain fart.